src/libs/actions/User.js

```diff
@@ -168,6 +174,38 @@ function validateLogin(accountID, validateCode) {
     });
 }
 
+/**
+ * Fetch the public domain info for the current user.
+ *
+ * This API is a bit weird in that it sometimes depends on information being cached in bedrock.
+ * If the info for the domain is not in bedrock, then it creates an asynchronous bedrock job to gather domain info.
+ * If that happens, this function will automatically retry itself in 10 minutes.
+ */
+function getPublicDomainInfo() {
+    // If this command fails, we'll retry again in 10 minutes,
+    // arbitrarily chosen giving Bedrock time to resolve the ClearbitCheckPublicEmail job for this email.
+    const RETRY_TIMEOUT = 600000;
+
+    // First check list of common public domains
+    if (_.contains(COMMON_PUBLIC_DOMAINS, sessionEmail)) {
+        Onyx.merge(ONYXKEYS.USER, {isFromPublicDomain: true});
+        return;
+    }
+
+    // If it is not a common public domain, check the API
+    API.User_IsFromPublicDomain({email: sessionEmail})
+        .then((response) => {
+            if (response.jsonCode === 200) {
+                const {isFromPublicDomain} = response;
+                Onyx.merge(ONYXKEYS.USER, {isFromPublicDomain});
+            } else {
+                // eslint-disable-next-line max-len
+                console.debug(`Command User_IsFromPublicDomain returned error code: ${response.jsonCode}. Most likely, this means that the domain ${Str.extractEmail(sessionEmail)} is not in the bedrock cache. Retrying in ${RETRY_TIMEOUT / 1000 / 60} minutes`);
```

Comment:
Not E.cash knowledgable, so not a blocker, is there a reason why we do `console.debug()` in e.cash comparing to our Web code.

Reply:
Well we can use [this library](https://github.com/Expensify/Expensify.cash/blob/7d73865ded6ba255752f06ef23ad7dbb867747c0/src/libs/Log.js) for server logs. Those should show up in LogSearch I believe. I just didn't think we needed this in server logs, but we could put them there. What do you think?

But if you're asking why we used `console.debug` instead of `console.log`, it's mostly because we have some ESLint rule that bans `console.log`. I'm honestly not too sure why.

Reply:
I don't know honestly, we don't usually log errors in the console and just keep them in the server, that's why I am asking. But maybe since now the repo is public, this makes easier for contributors to debug, no idea haha

Reply:
Well if this happens, we'll likely have server logs for the `ClearbitCheckPublicEmail` bedrock job that was just created, so let's just leave this for now. I think it'll maybe be useful for debugging, esp. since it's not obvious why this request might fail.